test/tm/generator.go

```diff
@@ -29,21 +30,43 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
+const (
+	defaultNetworkVPCCIDR      = "10.250.0.0/16"
+	defaultNetworkInternalCidr = "10.250.112.0/22"
+	defaultNetworkPublicCidr   = "10.250.96.0/22"
+	defaultNetworkWorkerCidr   = "10.250.0.0/19"
+)
+
 var (
-	infrastructureProviderConfigPath = flag.String("infrastructure-provider-config-filepath", "", "filepath to the provider specific infrastructure config")
-	controlplaneProviderConfigPath   = flag.String("controlplane-provider-config-filepath", "", "filepath to the provider specific controlplane config")
+	cfg    *GeneratorConfig
+	logger logr.Logger
+)
 
-	networkVPCCidr      = flag.String("network-vpc-cidr", "10.250.0.0/16", "vpc network cidr")
-	networkInternalCidr = flag.String("network-internal-cidr", "10.250.112.0/22", "internal network cidr")
-	networkPublicCidr   = flag.String("network-public-cidr", "10.250.96.0/22", "public network cidr")
-	networkWorkerCidr   = flag.String("network-worker-cidr", "10.250.0.0/19", "worker network cidr")
+type GeneratorConfig struct {
+	networkVPCCidr                   string
+	networkInternalCidr              string
+	networkPublicCidr                string
+	networkWorkerCidr                string
+	infrastructureProviderConfigPath string
+	controlplaneProviderConfigPath   string
+	zone                             string
+}
 
-	zone = flag.String("zone", "", "cloudprovider zone fo the shoot")
-)
+func init() {
```

Comment:
Can you move this to `main` or some function `addFlags` or similar?
I don't like the implicitness of `init` functions and try to use them only when necessary. Apart from that having it in a dedicated `addFlags` func will improve readability in general.